src/allmydata/test/no_network.py

```diff
@@ -188,8 +189,8 @@ def NoNetworkClient(basedir):
     # XXX FIXME this is just to avoid massive search-replace for now;
     # should be create_nonetwork_client() or something...
     from allmydata.node import read_config
-    config = read_config(basedir, u'client.port')
-    return _NoNetworkClient(config, basedir=basedir)
+    config = read_config(basedir, u'client.port', _valid_config_sections=client_valid_config_sections)
```

Comment:
Should `allmydata.client` expose a public `read_config` that takes care to pass in the valid client config sections?

Or maybe `allmydata.client.create_client` should accept an optional factory argument (defaulting to `_Client`) and then this code could use `create_client(basedir, _NoNetworkClient)`.

Reply:
My dream is that `Node`/`Client` itself will one day take enough args that we can pass in "no network" versions of (e.g.) the `Tub`s etc and then `NoNetworkClient` can be burned down. But, that day might be a while :/ so...